libs/core/src/lib/product-switch/product-switch-body/product-switch-body.component.ts

```diff
@@ -40,83 +46,96 @@ export class ProductSwitchBodyComponent implements OnInit {
     @Output()
     readonly itemClicked: EventEmitter<void> = new EventEmitter<void>();
 
-    /** Defines if the product switcher should be displayed in list mode, no matter what is the size of window */
-    @Input()
-    forceListMode = false;
+    /** @hidden */
+    private _istMode: boolean;
+
+    /** @hidden */
+    private _isRtl = false;
+
+    /** @hidden */
+    private _subscriptions = new Subscription();
 
-    /**
-     * @hidden
-     */
-    listMode: boolean;
+    constructor(@Optional() private readonly _rtlService: RtlService) {}
 
     /** @hidden */
     ngOnInit(): void {
+        this._subscriptions.add(this._rtlService?.rtl.subscribe((isRtl) => (this._isRtl = isRtl)));
+
+        this._checkSize();
+    }
+
+    ngOnDestroy(): void {
+        this._subscriptions.unsubscribe();
+    }
+
+    /** @hidden */
+    @HostListener('window:resize', [])
```

Comment:
Let's use cdk [viewport-ruller](https://github.com/angular/components/blob/master/src/cdk/scrolling/viewport-ruler.ts) instead, it gives throttling and runs out of angular zone so doesn't trigger extra change detection